test.js

```diff
@@ -185,3 +185,45 @@ test('ensure different promises are returned on second call with the same select
 	document.querySelector('.unicorn').remove();
 	t.is(prependElement(), await elementReady('.unicorn'));
 });
+
+test('ensure that the whole element has loaded', async t => {
+	const {window} = new JSDOM('<nav class="loading-html-fixture">');
+	const {document} = window;
+
+	// Fake the pre-DOM-ready state
+	const state = 'loading';
+	Object.defineProperty(document, 'readyState', {
+		get: () => state
+	});
+
+	let partialCheck;
+	(async () => {
+		partialCheck = await elementReady('nav', {
+			target: document,
+			expectEntireElement: false
+		});
+	})();
```

Comment:
Kinda ugly, I was hoping for https://github.com/sindresorhus/promise-fun/issues/12

Reply:
You can use `p-state` here now.